sdks/java/io/kinesis/src/test/java/org/apache/beam/sdk/io/kinesis/AmazonKinesisMock.java

```diff
@@ -163,6 +170,10 @@ public Record apply(@Nullable TestData testData) {
     public AmazonCloudWatch getCloudWatchClient() {
       return Mockito.mock(AmazonCloudWatch.class);
     }
+
+    @Override public IKinesisProducer getKinesisProducer(KinesisProducerConfiguration config) {
+      throw new RuntimeException("Not implemented");
+    }
   }
 
   private final List<List<Record>> shardedData;
```

Comment:
Doing UT on a mock is very weak because some problems appear only on real backends. And in those cases, we would like the build to fail. So having a embedded backend is better. If it is not possible, then you must test in IT to a real backend and make sure that the IT a scheduled on a frequent basis

Reply:
is there any embedded version of kinesis provided by Amazon for the tests?